src/kinetics/ReactionRateFactory.cpp

```diff
@@ -0,0 +1,99 @@
+ /**
+ *  @file ReactionRateFactory.cpp
+ */
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/kinetics/ReactionRateFactory.h"
+#include "cantera/kinetics/MultiRate.h"
+#include "cantera/thermo/ThermoPhase.h"
+#include "cantera/kinetics/Kinetics.h"
+#include "cantera/base/AnyMap.h"
+
+namespace Cantera
+{
+
+ReactionRateFactory* ReactionRateFactory::s_factory = 0;
+std::mutex ReactionRateFactory::rate_mutex;
+
+ReactionRateFactory::ReactionRateFactory()
+{
+    // ArrheniusRate evaluator
+    reg("ArrheniusRate", [](const AnyMap& node, const Units& rate_units) {
+        return new ArrheniusRate(node, rate_units);
+    });
+    addAlias("ArrheniusRate", "");
+    addAlias("ArrheniusRate", "elementary");
+    addAlias("ArrheniusRate", "three-body");
+
+    // PlogRate evaluator
+    reg("PlogRate", [](const AnyMap& node, const Units& rate_units) {
+        return new PlogRate(node, rate_units);
+    });
+    addAlias("PlogRate", "pressure-dependent-Arrhenius");
+
+    // ChebyshevRate evaluator
+    reg("ChebyshevRate", [](const AnyMap& node, const Units& rate_units) {
+        return new ChebyshevRate3(node, rate_units);
+    });
+    addAlias("ChebyshevRate", "Chebyshev");
```

Comment:
Do we need to introduce these new names like `ChebyshevRate` and `PlogRate`, or can we just use the names defined in the YAML format? I can see a case for handling `Arrhenius` differently, as having the canonical name be the empty string would be a little weird.

Reply:
This is an interesting proposition. While it muddles things a little short-term (no clear differentiation of names for `ReactionRate` and `Reaction` objects), it mostly avoids the `canonicalName` issue. Long-term, I don't believe that there will be a need to have different reaction types, as everything is determined by the attached rate object.

So yes, this is a great way to avoid naming issues down the road ...  :+1: